src/frontend-samples/display-styles-sample/DisplayStylesWidget.tsx

```diff
@@ -0,0 +1,82 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import "common/samples-common.scss";
+import React, { useEffect } from "react";
+import DisplayStylesApp from "./DisplayStylesApi";
+import { useActiveViewport } from "@bentley/ui-framework";
+import { displayStyles } from "./Styles";
+import { Select, Toggle } from "@bentley/ui-core";
+import { AbstractWidgetProps, StagePanelLocation, StagePanelSection, UiItemsProvider, WidgetState } from "@bentley/ui-abstract";
+
+const CUSTOM_STYLE_INDEX = 0;
+const DEFAULT_STYLE_INDEX = 4;
+
+export const DisplayStylesWidget: React.FunctionComponent = () => {
+  const viewport = useActiveViewport();
+  const [activePresetIndex, setActivePresetIndex] = React.useState<number>(DEFAULT_STYLE_INDEX);
+  const [mergeState, setMergeState] = React.useState<boolean>(false);
+
+  useEffect(() => {
+    if (viewport) {
+      let style = displayStyles[activePresetIndex];
+      DisplayStylesApp.applyDisplayStyle(viewport, style);
+
+      if (mergeState) {
+        style = displayStyles[CUSTOM_STYLE_INDEX];
+        DisplayStylesApp.applyDisplayStyle(viewport, style);
+      }
+    }
+  }, [activePresetIndex, mergeState, viewport]);
```

Comment:
It's now unnecessary to specify use effect decencies.  Please remove unless there is another reason.

Reply:
The useEffect depends on all three of those dependencies. Its best practice to have them there.